bcycle-backend/Controllers/TripsController.cs

```diff
@@ -1,109 +1,74 @@
-            this._dbContext = dbContext;
+            _tripService = tripService;
         }
 
-
-        // temporary testing placeholder
-        [HttpPost("fakeuser")]
-        public async Task<ActionResult> CreateFakeUser()
-        {
-            User user = await UserDataHelper.GetCurrentUserDetails();
-            _dbContext.Add(user);
-            await _dbContext.SaveChangesAsync();
-            return Ok();
-        }
-
-        // GET api/trips
+        // GET /api/trips
         [HttpGet]
         public async Task<ActionResult<ResultContainer<IEnumerable<TripDto>>>> Get()
         {
-            int userID = (await UserDataHelper.GetCurrentUserID()).Value;
-            return new ResultContainer<IEnumerable<TripDto>>(
-                await _dbContext.Trips
-                    .Where(t => t.UserID == userID)
-                    .Include(t => t.TripPhotos).Include(t => t.TripPoints)
-                    .OrderByDescending(t => t.Started)
-                    .Select(t => t.AsDto())
-                    .ToListAsync());
+            var trips = await _tripService.GetAll(User.GetId()).ToListAsync();
+            return new ResultContainer<IEnumerable<TripDto>>(trips);
         }
 
-        // GET api/trips/5
+        // GET /api/trips/{id}
         [HttpGet("{id}")]
         public async Task<ActionResult<ResultContainer<TripDto>>> Get(int id)
         {
-            return new ResultContainer<TripDto>((await _dbContext.GetMyTrip(id)).AsDto());
+            var trip = await _tripService.GetUserTripAsync(id, User.GetId());
+            if (trip == null)
+                return NotFound();
+            
+            return new ResultContainer<TripDto>(trip.AsDto());
         }
 
-        // POST api/trips
+        // POST /api/trips
         [HttpPost]
         public async Task<ActionResult<ResultContainer<int>>> Post([FromBody] TripDto tripDto)
         {
-            tripDto.ID = 0;
-            Trip trip = tripDto.AsTrip();
-            trip.UserID = (await UserDataHelper.GetCurrentUserID()).Value;
-            
-            _dbContext.Add(trip);
-            await _dbContext.SaveChangesAsync();
-            return new ResultContainer<int>(trip.ID);
+            var savedTrip = await _tripService.SaveTripAsync(tripDto, User.GetId());
+            return new ResultContainer<int>(savedTrip.Id);
         }
 
-        // PUT api/trips/5/photo
+        // PUT /api/trips/{id}/photo
         [HttpPut("{id}/photo")]
         public async Task<ActionResult<ResultContainer<string>>> PutPhoto(int id)
         {
-            Trip trip = await _dbContext.GetMyTrip(id);
-
-            string uploadPath = _configuration.GetValue<string>("UploadPath");
-            string uploadPrefix = _configuration.GetValue<string>("UploadPrefix");
-            Directory.CreateDirectory(uploadPath);
+            var photoData =
+                new CreatePhotoData {Payload = Request.Body, UrlBase = $"{Request.Scheme}://{Request.Host}"};
```

Comment:
Why add an extra class? What’s wrong with just passing those two values as parameters to PutPhotoAsync?

Reply:
Idk you may be right. 4 params seemed ugly

Reply:
Done